core/google/cloud/iterator.py

```diff
@@ -378,9 +379,19 @@ def _get_next_page_response(self):
         :rtype: dict
         :returns: The parsed JSON response of the next page's contents.
         """
-        return self.client.connection.api_request(
-            method='GET', path=self.path,
-            query_params=self._get_query_params())
+        params = self._get_query_params()
+        if self._HTTP_METHOD == 'GET':
```

Comment:
Not really. The `HTTPIterator` makes a "list" type HTTP request. Previously it was **always** a GET request but now it is **sometimes** a GET request. The base connection doesn't do anything other than `api_request` and it's up to the caller to provide the HTTP method.